administrator/components/com_workflow/Model/StatesModel.php

```diff
@@ -85,6 +86,20 @@ protected function populateState($ordering = null, $direction = null)
 	}
 
 	/**
+	 * A protected method to get a set of ordering conditions.
+	 *
+	 * @param   object  $table  A record object.
+	 *
+	 * @return  array  An array of conditions to add to add to ordering queries.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	protected function getReorderConditions($table)
+	{
+		return 'extension = ' . $this->getDbo()->q($table->extension);
```

Comment:
Here you define the condition, within which group the ordering should happen, so I guess the correct "condition" would be ```workflow_id```

Reply:
ops, there should be both, because we need extension too :) right now i'm not sure where we have used it but for sure it was :D